tests/threadname_test.cc

```diff
@@ -0,0 +1,30 @@
+#include "gtest/gtest.h"
+#include <pistache/http.h>
+#include <pistache/endpoint.h>
+
+using namespace Pistache;
+
+void setThreadNameTest(const std::string pThreadName)
+{
+    std::shared_ptr<Http::Endpoint> mHttpEndpoint_;
+    Address addr(Ipv4::any(), Pistache::Port(0));
+    mHttpEndpoint_ = std::make_shared<Http::Endpoint>(addr);
+
+    auto test_options = Http::Endpoint::options()
+          .threads(2)
+          .threadsName(pThreadName);
+    mHttpEndpoint_->init(test_options);
+}
```

Comment:
The actual call to pthread_setname_np() is inside Worker::run().  Is that code path exercised via this test?  Hint: temporarily add a call to "abort()" right after pthread_setname_np(), then run this test.  If the test does not crash, then you're not testing the call all the way down to the bottom of the stack.  You might need to extend this test to actually start the server.  Of course, it is also possible that your code does already test your change - I'm only eye-balling your PR, I've not actually cloned it to my local workstation and tested it myself (yet).